### ISSUE TITLE
Solving a "Mixed content" problem

### ISSUE THREAD

**Issue by stevejburge** [publishpress/PublishPress#438]
A couple of users have reported problems with "mixed content" errors

@agapetry took a look and report:

 

> I think that error could be caused by the plugin loading scripts or stylesheets with an http URL, or by outputting an http link which the user follows. From a scan of search results, I see the following possible trouble spots. In particular, the http links from Base.php (below) could cause this problem if the user's site isn't forcing http via .htaccess rewrite rule or redirect (and could also explain why we don't see the error).
> 
> libraries\Notifications\Table\Base.php
> ('http://'.$_SERVER['HTTP_HOST'] in function pagination, function print_column_headers)
> 
> modules\efmigration\lib\js\react-dom.min.js
> (http://www.w3.org in various functions)
> 
> vendor\alledia\wordpress-plugin-framework\src\assets\lib\font-awesome-v5.2.0\js\all.js
> (http://www.w3.org in function makeInlineSvgAbstract)
> 
> vendor\alledia\wordpress-plugin-framework\src\assets\lib\font-awesome-v5.2.0\js\fontawesome.js
> (http://www.w3.org in function makeInlineSvgAbstract)
> 
> vendor\alledia\wordpress-plugin-framework\src\assets\lib\react-v16.4.2\react-dom.development.js
> (http://www.w3.org in function shouldRemoveAttribute, function restoreControlledState$3

**Comment by stevejburge** [comment]
Here's another customer. The problem was with cdn.jsdelivr.net/select2/3.5.2/select2.min.js?ver=3.5.2

![Screen Shot 2019-06-20 at 12 12 28 PM](https://user-images.githubusercontent.com/3868028/59864875-bc2bc880-9355-11e9-9419-3ce81df1da11.png)

**Comment by stevejburge** [comment]
@andergmartins This will soon become a bigger problem: https://itknowing.com/google-chrome-will-block-mixed-content-in-the-near-future

**Comment by andergmartins** [comment]
Thanks @stevejburge If we still have this issue (I'm double checking) we already have problems because they already block scripts and stylesheets. The future changes are for audio/video and image resources, still accepted as mixed content.

**Comment by andergmartins** [comment]
Fixed the url protocol in the Base table library, but the reports related to 3rd party JS libraries are not really a problem. I checked the new versions of those libraries and they still define xmlns URLs using http only. If we decide to change this I would advise to open Github issues for each library and ask them to fix in the core. But since those URLs are not actually assets loaded in the page, I don't think they could cause problems. Probably a false positive.
